platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/StaticTopology.java

```diff
@@ -98,20 +84,15 @@ public boolean shouldConnectToMe(final NodeId nodeId) {
      * @return true if this node is my neighbor, false if not
      */
     private boolean isNeighbor(final NodeId nodeId) {
-        if (!addressBook.contains(nodeId)) {
-            return false;
-        }
-        final int selfIndex = addressBook.getIndexOfNodeId(selfId);
-        final int nodeIndex = addressBook.getIndexOfNodeId(nodeId);
-        return connectionGraph.isAdjacent(selfIndex, nodeIndex);
+        return peerNodeToIdMap.containsKey(nodeId);
```

Comment:
You've dropped the `isAdjacent` check with connectionGraph.   If the PR is just to remove the address book dependency then you still need to query the index from the map you construct once it has the right value in the key-value pair.

Reply:
outdated. Resolving.